models/CMakeLists.txt

```diff
@@ -1,19 +1,30 @@
-# List the subdirectory (http://stackoverflow.com/questions/7787823/cmake-how-to-get-the-name-of-all-subdirectories-of-a-directory)
-macro(SUBDIRLIST result curdir)
-  file(GLOB children RELATIVE ${curdir} ${curdir}/*)
-  set(dirlist "")
-  foreach(child ${children})
-    if(IS_DIRECTORY ${curdir}/${child})
-      list(APPEND dirlist ${child})
-    endif()
-  endforeach()
-  set(${result} ${dirlist})
-endmacro()
-
-# Get list of models 
-subdirlist(subdirs ${CMAKE_CURRENT_SOURCE_DIR})
-
-# Install each model 
-foreach (dir ${subdirs})
-  install (DIRECTORY ${dir} DESTINATION share/gazebo/models)
-endforeach ()
\ No newline at end of file
+# configure_files for all files in a folder
+# https://stackoverflow.com/questions/697560/how-to-copy-directory-from-source-tree-to-binary-tree
+macro(CONFIGURE_FILES srcDir destDir)
+  make_directory(${destDir})
+  file(GLOB_RECURSE templateFiles RELATIVE ${srcDir} ${srcDir}/*)
+
+  foreach(templateFile ${templateFiles})
+    set(srcTemplatePath ${srcDir}/${templateFile})
+    message(${srcTemplatePath})
+
+    if(NOT IS_DIRECTORY ${srcTemplatePath})
+      configure_file(${srcTemplatePath}
+                     ${destDir}/${templateFile}
+                     @ONLY)
+    endif()
+
+  endforeach(templateFile)
+
+endmacro(CONFIGURE_FILES)
+
+# Allow specifying the robot name
+set(ROBOT_NAME "" CACHE STRING "Select the the robot name that will match YARP_ROBOT_NAME")
+
+# Check if YARP_ROBOT_NAME is defined as env variable
+if(ROBOT_NAME STREQUAL "")
+  message(WARNING "ROBOT_NAME option is not defined. Using icub as default for backwards compatibility")
```

Comment:
Sorry for the delay in checking this. It seems to me that `icub` is not set as the default, even if the warnings says otherwise?

Reply:
I completely forgot to finalize this PR. As discussed in https://github.com/robotology-playground/icub-gazebo-wholebody/pull/12#issuecomment-318979158 the default will be `icub`.